app/main/views/drafts.py

```diff
@@ -36,16 +39,19 @@ def create_draft_service(service_id):
         abort(400, "Draft already exists for service {}".format(service_id))
 
     draft = DraftService.from_service(service)
+
+    db.session.add(draft)
+    db.session.flush()
```

Comment:
Flush can trigger an `IntegrityError`, so it might be worth wrapping it with `try ... except`